### PR TITLE
esp32_gpio: allow to set GPIO level from outside

### PR DESCRIPTION
**Pull request by knw1**

### CODE REVIEW

File: hw/gpio/esp32_gpio.c [hunk 1]
```diff
@@ -19,26 +19,197 @@
...
+#include "qapi/qmp/qdict.h"
+#include "qapi/qmp/qjson.h"
+#include "qapi/qmp/qnum.h"
 
+// #define ENABLE_DEBUG
+
+#ifdef ENABLE_DEBUG
+#define D(x) x
+#else
+#define D(x)
```
Comment (dmsspyro) [inline hw/gpio/esp32_gpio.c]: Nitpicking: `#define D(x) do {} while(0)` is a little bit less error-prone. I mean in current solution we can write something like that:
```{.c}
D(int a = 0) // lack of semicolon
D(a++) // lack of semicolon
```
and it's be perfectly fine as long as debug is disabled, but will fail during compilation with debug enabled..

The `do-while` approach fails either way

**Review by dmsspyro — APPROVED** [review]
For me it looks well (only one not-so-important side note)

File: hw/gpio/esp32_gpio.c (hunk 1 above)
Comment (knw1) [reply hw/gpio/esp32_gpio.c]: Corrected

**Review by dmsspyro — APPROVED** [review]
New package of changes reviewed. Looks good to me.

**Pull request merged**